### PR TITLE
Implement basic swerve sim

### PR DESCRIPTION
**Pull request by 3LucasZ**

### CODE REVIEW

**Comment by realjoshuau** [comment]
Two things before I can merge (pre-review):
1. Delete `BuildConstants.java` & add it to your .gitignore
2. Make **sure** you run `./gradlew spotlessCheck` and `./gradlew spotlessApply` before you request a review.

File: src/main/java/frc/robot/subsystems/swerve/SwerveDrive.java [hunk 1]
```diff
@@ -68,7 +75,10 @@ public void setModuleStates(SwerveModuleState[] desiredStates) {
   }
 
   public Pose2d getPose() {
-    return swerveOdometry.getPoseMeters();
+    if (Robot.isReal())
```
Comment (realjoshuau) [inline src/main/java/frc/robot/subsystems/swerve/SwerveDrive.java]: Nice job here checking for both sim & real mode.

File: src/main/java/frc/robot/subsystems/swerve/SwerveDrive.java [hunk 2]
```diff
@@ -11,46 +11,53 @@
...
 import frc.robot.Constants;
+import frc.robot.Robot;
 
 public class SwerveDrive extends SubsystemBase {
   public SwerveDriveOdometry swerveOdometry;
   public SwerveModule[] mSwerveMods;
   public Pigeon2 gyro;
+  public Pose2d simSwervePose;
+  public ChassisSpeeds curChassisSpeed;
+  private final Field2d field = new Field2d();
```
Comment (realjoshuau) [inline src/main/java/frc/robot/subsystems/swerve/SwerveDrive.java:34]: nitpicking: does this _need_ to be private?

*no need to change anything for this

**Review by realjoshuau — DISMISSED** [review]
Nice! Code is clean & good. Great job on this PR @3LucasZ & @DragonXDev 

*Still need to fix merge conflicts.

**Review by DragonXDev — DISMISSED** [review]
Looks good, the sim Pose2d(); seems to work perfect here, and the sim looks great.

File: src/main/java/frc/robot/subsystems/swerve/SwerveDrive.java (hunk 2 above)
Comment (3LucasZ) [reply src/main/java/frc/robot/subsystems/swerve/SwerveDrive.java]: Okay, cool.

**Review by DragonXDev — APPROVED** [review]
Looks good.

**Pull request merged**